data_steward/vocabulary.py

```diff
@@ -89,30 +94,61 @@ def transform_files(in_dir, out_dir):
         transform_file(in_path, out_dir)
 
 
-def get_aou_general_version():
-    return hash_dir(AOU_GENERAL_PATH)
+def get_aou_vocab_version():
+    """
+    Generate an identifier used to version AOU vocabulary resources
+
+    :return: unique identifier string
+    """
+    return hash_dir(AOU_VOCAB_PATH)
 
 
-def get_aou_general_vocabulary_row():
-    aou_gen_version = get_aou_general_version()
+def get_aou_vocabulary_row(vocab_id):
+    """
+    Get row for the vocabulary
+
+    :param vocab_id:  vocabulary id to generate row for
+    :return: a delimited string representing row of the vocabulary.csv file
+    """
+    aou_vocab_version = get_aou_vocab_version()
     # vocabulary_id vocabulary_name vocabulary_reference vocabulary_version vocabulary_concept_id
-    return DELIMITER.join([AOU_GEN_ID, AOU_GEN_NAME, AOU_GEN_VOCABULARY_REFERENCE, aou_gen_version,
-                           AOU_GEN_VOCABULARY_CONCEPT_ID])
+    vocab_row = VOCABULARY_UPDATES.get(vocab_id)
+    vocab_row[-2] = aou_vocab_version
+    return DELIMITER.join(vocab_row)
+
+
+def _vocab_id_match(s):
+    """
+    Get a matching AOU vocabulary ID in the specified string, if any
+
+    :param s: string to search for AOU vocabulary IDs
+    :return: the first vocabulary ID found in the string, otherwise None
+    """
+    vocab_id_in_row_iter = (vocab_id for vocab_id in VOCABULARY_UPDATES if vocab_id in s)
+    return next(vocab_id_in_row_iter, None)
```

Comment:
This is interesting.  I haven't used `next()` before.  I had to run it to see the difference in returning the iterator value vs. the list.  A comment might make what is happening clearer for new developers.